watchtower-plugin/src/retrier.rs

```diff
@@ -0,0 +1,682 @@
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use tokio::sync::mpsc::UnboundedReceiver;
+
+use backoff::future::retry_notify;
+use backoff::{Error, ExponentialBackoff};
+
+use teos_common::cryptography;
+use teos_common::errors;
+use teos_common::UserId as TowerId;
+
+use crate::net::http::{add_appointment, AddAppointmentError};
+use crate::wt_client::WTClient;
+use crate::AppointmentStatus;
+
+pub async fn manage_retry(
+    mut unreachable_towers: UnboundedReceiver<TowerId>,
+    wt_client: Arc<Mutex<WTClient>>,
+    max_elapsed_time_secs: u16,
+) {
+    log::info!("Starting retry manager");
+
+    loop {
+        let tower_id = unreachable_towers.recv().await.unwrap();
+        wt_client
+            .lock()
+            .unwrap()
+            .set_tower_status(tower_id, crate::TowerStatus::TemporaryUnreachable);
+
+        log::info!("Retrying tower {}", tower_id);
+        match retry_notify(
+            ExponentialBackoff {
+                max_elapsed_time: Some(Duration::from_secs(max_elapsed_time_secs as u64)),
+                ..ExponentialBackoff::default()
+            },
+            || async { retry_add_appointment(tower_id, wt_client.clone()).await },
+            |err, _| {
+                log::warn!("Retry error happened with {}. {}", tower_id, err);
+            },
+        )
+        .await
+        {
+            Ok(_) => {
+                log::info!("Retry strategy succeeded for {}", tower_id);
+                wt_client
+                    .lock()
+                    .unwrap()
+                    .set_tower_status(tower_id, crate::TowerStatus::Reachable);
+            }
+            Err(e) => {
+                log::warn!("Retry strategy gave up for {}. {}", tower_id, e);
+                if wt_client
```

Comment:
It can also be misbehaving. The retry strategy gives up when a persistent error is raised. That can happen either manually (by returning `Err(Error::permanent(...))` or when the the backoff finishes without a sucess.

I guess it'd be at least worth documenting on top of that condition that `TowerStatus::Misbehaving` is also an expected status at this point.